src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -274,13 +274,19 @@ private CallbackResponse aboutToSubmit(CallbackParams callbackParams) {
             builder.respondent1ClaimResponseDocumentSpec(null);
         }
 
-        locationHelper.getCaseManagementLocation(caseData)
-            .ifPresent(requestedCourt -> locationHelper.updateCaseManagementLocation(
-                builder,
-                requestedCourt,
-                () -> locationRefDataService.getCourtLocationsForDefaultJudgments(callbackParams.getParams().get(
-                    CallbackParams.Params.BEARER_TOKEN).toString())
-            ));
+        //Update the caseManagement location to the flight location if No flight location update to Claimant
+        // preferred location
+        if (isFlightDelayClaimAndUpdateCaseLocation(caseData)) {
```

Comment:
I think this condition should be revised. What will happen when SDOR2 go live? Code should work wether SDOR2 is true or not. This code will not work if SDOR2 is true and its not flightDelay claim.